cmd/kube-scheduler/app/server.go

```diff
@@ -58,6 +58,8 @@ import (
 	"k8s.io/klog"
 )
 
+var frameworkRegistry = framework.NewRegistry()
```

Comment:
This does not look the right way of creating registry to me. We should not create a global singleton. Instead, we should pass a parameter. Looking at the code, I think we need to make changes to more than one function in server.go to make  passing a registry from an outside package possible.

Reply:
Since the scheduler is singleton, maybe it be reasonable that makes the registry a singleton? If you do not agree with it, I'll modify it to pass a parameter.

Reply:
+1, I mentioned that in my initial review as well, we shouldn't be using global variables like that. The client should either create a registry and pass it, or use the With pattern I mentioned earlier.

I understand that there is no precedence to passing arguments when creating a *Command in k8s, but that does not mean it is wrong, it is just that no other command require the type of customization that we want to support in the scheduler.

@bsalamat why do we need to make changes to more than one function? isn't enough to just allow NewSchedulerCommand accept a registry parameter?

Reply:
Addressed. Add a new function `NewSchedulerCommandWithRegistry` for users. Then users could use it this way:

```
   registry := framework.NewRegistry()
   registry.Add("MyPlugin", NewMyPlugin)
   scheduler := NewSchedulerCommandWithRegistry(registry)
   ...
```

@bsalamat @ahg-g PTAL